main.nf

```diff
@@ -191,67 +235,655 @@ process get_software_versions {
+  msspsmtable specdata -i filtpep --dbfile psmlookup -o prepsms.txt
+  ${!params.noquant ? "msspsmtable quant -i prepsms.txt -o qpsms.txt --dbfile psmlookup --precursor ${params.isobaric && td=='target' ? '--isobaric' : ''}" : 'mv prepsms.txt qpsms.txt'}
+  sed 's/\\#SpecFile/SpectraFile/' -i qpsms.txt
+  ${!params.onlypeptides ? "msspsmtable genes -i qpsms.txt -o gpsms --dbfile psmlookup" : ''}
+  ${!params.onlypeptides ? "msslookup proteingroup -i qpsms.txt --dbfile psmlookup" : ''}
+  ${!params.onlypeptides ? "msspsmtable proteingroup -i gpsms -o pgpsms --dbfile psmlookup" : 'mv qpsms.txt pgpsms'}
+  ${params.hirief ? "peptide_pi_annotator.py -i $trainingpep -p pgpsms --o dppsms --stripcolpattern Strip --pepcolpattern Peptide --fraccolpattern Fraction --strippatterns ${allstrips.join(' ')} --intercepts ${allstrips.collect() { params.strips[it].intercept}.join(' ')} --widths ${allstrips.collect() { params.strips[it].fr_width}.join(' ')} --ignoremods \'*\'" : ''}
+  msspsmtable split -i ${params.hirief ? 'dppsms' : 'pgpsms'} --bioset
+  mv ${params.hirief ? 'dppsms' : 'pgpsms'} ${td}_psmtable.txt
+  mv psmlookup ${td}_psmlookup.sql
+  """
+}
+
+// Collect setnames and merge with PSM tables for peptide table creation
+setnames_psm
+  .toList()
+  .map { it -> [it.sort()]}
+  .set { setlist_psm }
+setpsmtables
+  .map { it -> [it[0], it[1] instanceof java.util.List ? it[1] : [it[1]] ] }
+  .map{ it -> [it[0], it[1].sort { a, b -> a.baseName.tokenize('.')[0] <=> b.baseName.tokenize('.')[0] }] } // names are setnames, sort on them then merge with sorted setnames
+  .merge(setlist_psm)
+  .transpose()
+  .set { psm_pep }
+
+
+process psm2Peptides {
+
+  input:
+  set val(td), file('psms'), val(setname) from psm_pep
+  
+  output:
+  set val(setname), val(td), file("${setname}_linmod"), file('proteinratios') into pepslinmod
+  set val(setname), val('peptides'), val(td), file("${setname}_linmod") into peptides_out
+  set val(setname), file('normratiosused') optional true into normratios
+  set val(setname), val(td), file(psms), file('proteins'), val('proteins') into proteins
+  set val(setname), val(td), file(psms), file('genes'), val('genes') into genes
+  set val(setname), val(td), file(psms), file('symbols'), val('assoc') into symbols
+  set val(setname), file('proteinratios') optional true into proteinratios
+
+  script:
+  col = accolmap.peptides + 1  // psm2pep adds a column
+  """
+  msspeptable psm2pep -i psms -o peptides --scorecolpattern svm --spectracol 1 ${!params.noquant && params.isobaric && td == 'target' ? "--isobquantcolpattern plex" : "" } ${!params.noquant ? "--ms1quantcolpattern area" : ""}
+  paste <( cut -f ${col} peptides) <( cut -f 1-${col-1},${col+1}-500 peptides) > peptide_table.txt
+  echo Protein accession |tee proteins genes symbols
+  tail -n+2 psms|cut -f ${accolmap.proteins}|grep -v '\\;'|grep -v "^\$"|sort|uniq >> proteins
```

Comment:
Especially this part - I get what the command does, but *why* you are doing this isn't clear to me :-)